presto-docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -168,58 +168,76 @@ Configuration Properties
+                                                        is required if the iceberg.catalog.type is ``hadoop``.
+                                                        Otherwise, it will be ignored.
+
+``iceberg.file-format``                                 The storage file format for Iceberg tables. The available     ``ORC``
+                                                        values are ``PARQUET`` and ``ORC``.
+
+``iceberg.compression-codec``                           The compression codec to use when writing files. The          ``GZIP``
+                                                        available values are ``NONE``, ``SNAPPY``, ``GZIP``,
+                                                        ``LZ4``, and ``ZSTD``.
+
+``iceberg.max-partitions-per-writer``                   The Maximum number of partitions handled per writer.          ``100``
+
+``iceberg.minimum-assigned-split-weight``               A decimal value in the range (0, 1] is used as a minimum      ``0.05``
+                                                        for weights assigned to each split. A low value may improve
+                                                        performance on tables with small files. A higher value may
+                                                        improve performance for queries with highly skewed
+                                                        aggregations or joins.
+
+``iceberg.enable-merge-on-read-mode``                   Enable reading base tables that use merge-on-read for         ``true``
+                                                        updates.
+
+``iceberg.delete-as-join-rewrite-enabled``              When enabled, equality delete row filtering is applied        ``true``
+                                                        as a join with the data of the equality delete files.
+
+``iceberg.enable-parquet-dereference-pushdown``         Enable parquet dereference pushdown.                          ``true``
+
+``iceberg.hive-statistics-merge-strategy``              Determines how to merge statistics that are stored in the     ``NONE``
+                                                        Hive Metastore. The available values are ``NONE``,
+                                                        ``USE_NULLS_FRACTION_AND_NDV``, ``USE_NULLS_FRACTIONS``
+                                                        and, ``USE_NDV``
+
+``iceberg.statistic-snapshot-record-difference-weight`` The amount that the difference in total record count matters
+                                                        when calculating the closest snapshot when picking
+                                                        statistics. A value of 1 means a single record is equivalent
+                                                        to 1 millisecond of time difference.
+
+``iceberg.pushdown-filter-enabled``                     Experimental: Enable filter pushdown for Iceberg. This is     ``false``
```

Comment:
Is this true?  I thought it worked with ORC files on Java workers.

Reply:
As per I am aware of this feature flag only make the Plan changes that are needed for Native as of now. There was limited support for iceberg table filter pushdown was added as part of https://github.com/prestodb/presto/pull/16964 which is available by default as it is, feature flag is not required.
@imjalpreet Are you aware of ORC files support?

Reply:
As I know, this property is indeed only for Native worker now. I also wonder whether we could use the same filter pushdown logic for both java workers and native workers? I'm not familiar with native iceberg connector implementation, are there irreconcilable logics between the two?

Reply:
I'll merge for now, but I think it should work for ORC since this just delegates to Aria scan (select readers) for ORC.  We can clarify in a follow up PR.  CC: @yingsu00 @imjalpreet

Reply:
In the current implementation, if this flag is set in the case of a Java worker, we fail the query and throw an exception that it is not supported irrespective of Parquet or ORC. The reason was we have not yet verified the current implementation when using ORC with Java Worker. The initial aim was to keep the Java code flow the same for now and use the new optimizer rule only in the case of a native worker. I can re-visit this once I wrap up and reach a stable state for Iceberg on Prestissimo. We can verify whether the current changes are sufficient for ORC with Java Worker or do we need any modifications. Based on that, we can change when we throw the exception.